clients/src/test/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManagerTest.java

```diff
@@ -1122,6 +1156,18 @@ public void testUnknownTopicIdError() {
         assertEquals(0L, metadata.timeToNextUpdate(time.milliseconds()));
     }
 
+    @Test
+    public void testDisconnectExceptionOnShareAcknowledge() {
+        buildRequestManager();
+        assignFromSubscribed(singleton(tp0));
+
+        assertEquals(1, sendFetches());
+        client.prepareResponse(fullFetchResponse(tip0, records, acquiredRecords, Errors.NONE));
+
```

Comment:
This doesn't look entirely finished. Either you're missing some additional test code that you omitted, or you've left a bunch of blank lines.